avm/res/azure-stack-hci/cluster/tests/e2e/2nodeswitched.defaults/main.test.bicep

```diff
@@ -0,0 +1,304 @@
+    qosPolicyOverrides: {
+      bandwidthPercentage_SMB: '50'
+      priorityValue8021Action_Cluster: '7'
+      priorityValue8021Action_SMB: '3'
+    }
+    overrideVirtualSwitchConfiguration: false
+    virtualSwitchConfigurationOverrides: {
+      enableIov: 'true'
+      loadBalancingAlgorithm: 'Dynamic'
+    }
+    trafficType: ['Management']
+  }
+  {
+    adapter: ['comp0', 'comp1']
+    name: 'compute'
+    overrideAdapterProperty: true
+    adapterPropertyOverrides: {
+      jumboPacket: '9014'
+      networkDirect: 'Disabled'
+      networkDirectTechnology: 'iWARP'
+    }
+    overrideQosPolicy: false
+    qosPolicyOverrides: {
+      bandwidthPercentage_SMB: '50'
+      priorityValue8021Action_Cluster: '7'
+      priorityValue8021Action_SMB: '3'
+    }
+    overrideVirtualSwitchConfiguration: false
+    virtualSwitchConfigurationOverrides: {
+      enableIov: 'true'
+      loadBalancingAlgorithm: 'Dynamic'
+    }
+    trafficType: ['Compute']
+  }
+  {
+    adapter: ['smb0', 'smb1']
+    name: 'storage'
+    overrideAdapterProperty: true
+    adapterPropertyOverrides: {
+      jumboPacket: '9014'
+      networkDirect: 'Disabled'
+      networkDirectTechnology: 'iWARP'
+    }
+    overrideQosPolicy: true
+    qosPolicyOverrides: {
+      bandwidthPercentage_SMB: '50'
+      priorityValue8021Action_Cluster: '7'
+      priorityValue8021Action_SMB: '3'
+    }
+    overrideVirtualSwitchConfiguration: false
+    virtualSwitchConfigurationOverrides: {
+      enableIov: 'true'
+      loadBalancingAlgorithm: 'Dynamic'
+    }
+    trafficType: ['Storage']
+  }
+]
+@description('Optional. Enable storage auto IP configuration. If false, storageNetworks must include IP configurations.')
+param enableStorageAutoIp bool = true
+@description('Optional. The storage networks for the cluster.')
+param storageNetworks storageNetworksArrayType = [
+  {
+    adapterName: 'smb0'
+    vlan: '711'
+  }
+  {
+    adapterName: 'smb1'
+    vlan: '712'
+  }
+]
+param deployArcGateway bool = false
+
+#disable-next-line no-hardcoded-location // Due to quotas and capacity challenges, this region must be used in the AVM testing subscription
+var enforcedLocation = 'southeastasia'
+
+var clusterWitnessStorageAccountName = '${take('${deploymentPrefix}${serviceShort}${take(uniqueString(resourceGroup.id,resourceGroup.location),6)}',21)}wit'
+var keyVaultDiagnosticStorageAccountName = '${take('${deploymentPrefix}${serviceShort}${take(uniqueString(resourceGroup.id,resourceGroup.location),6)}',21)}kvd'
+var keyVaultName = 'kvhci-${deploymentPrefix}${take(uniqueString(resourceGroup.id,resourceGroup.location),6)}'
+
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2024-03-01' = {
+  name: resourceGroupName
+  location: enforcedLocation
+}
+
+module hciDependencies 'dependencies.bicep' = {
+  name: '${uniqueString(deployment().name, enforcedLocation)}-test-hcidependencies-${serviceShort}'
+  scope: resourceGroup
+  params: {
+    clusterNodeNames: clusterNodeNames
```

Comment:
All names of top-level resources should be declared here with a `dep-` prefix and using the namePrefix & serviceShort values to make sure no 2 contributors & test cases deploy the same resources.
Also, only top-level service names should be passed down as they could conflict. All other values should be coded directly into the dependencies.bicep file (either as a variable, or even better directly where they're used). If any of the values are needed by the subsequent testDeployment, the `dependencies.bicep` file should return the needed values as outputs that are subsequently referenced.